core/src/types/transaction/mod.rs

```diff
@@ -33,6 +36,119 @@ use crate::types::ibc::IbcEvent;
 use crate::types::storage;
 use crate::types::transaction::protocol::ProtocolTx;
 
+/// The different result codes that the ledger may send back to a client
+/// indicating the status of their submitted tx
+#[derive(
+    Debug,
+    Copy,
+    Clone,
+    FromPrimitive,
+    ToPrimitive,
+    PartialEq,
+    Eq,
+    Serialize,
+    Deserialize,
+)]
+pub enum ResultCode {
```

Comment:
nit: maybe document that these codes shouldn't be changed around between versions, so people can rely on their integer values to identify errors

Reply:
done in [c0fc84b](https://github.com/anoma/namada/pull/2276/commits/c0fc84bd272cb0b4904d4440444018f543ab36c7)